similar/main/state.cpp

```diff
@@ -1536,6 +1536,12 @@ int state_restore_all_sub(const char *filename, const secret_restore secret)
 				do_cloak_invul_secret_stuff(old_gametime, ret_pl_info);
 			} else {
 				get_local_player() = dummy_player;
+				// Keep keys even if they died on secret level (otherwise game becomes impossible)
+				// Example: Cameron 'Stryker' Fultz's Area 51
+				pl_info.powerup_flags |= (plrobj.ctype.player_info.powerup_flags &
```

Comment:
This seems like a semantic change from 0.58.1.  This is in the `!= secret_restore::survived` path, so it seems like we ought to remove the keys obtained on the secret level and use only the keys from the save game.  In 0.58.1, this branch overwrote the active player with `dummy_player` and did nothing else; in that version, key flags were stored in `struct player`, so player keys would have reverted to whatever was in `dummy_player`, which had just been read from the secret save.  Did 0.58.1 really allow the player to keep keys from a secret level after dying there?  If not, why are there levels that require that behavior in order to let the player win?